src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -364,7 +364,8 @@ private static IUnboundLambdaExpression CreateUnboundLambdaOperation(UnboundLamb
             // an IUnboundLambdaExpression, you need to look at the parent IConversionExpression.
             ITypeSymbol type = null;
             Optional<object> constantValue = ConvertToOptional(unboundLambda.ConstantValue);
-            return new UnboundLambdaExpression(isInvalid, syntax, type, constantValue);
+            Lazy<ILambdaExpression> internalLambda = new Lazy<ILambdaExpression>(() => CreateBoundLambdaOperation(unboundLambda.BindForErrorRecovery()));
```

Comment:
>unboundLambda.BindForErrorRecovery() [](start = 114, length = 36)

It feels like we might want to delegate this work to SemanticModel, or coordinate it with SemanticModel. We don't want SemanticModel cache for bound nodes to somehow end up with a different bound nodes for the same lambda body. The recovery should exactly match the recovery performed by SemanticModel, it is better to reuse the same code across both components in order to ensure the logic is the same and won't diverge in the future. It is very likely that SemanticModel cache already contains a BoundLambda for this node and we just need to get it from there. Note, SemanticModel implementation could differ between languages.